src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -2049,22 +2125,18 @@ private boolean notifyCommit(FreeStyleProject project, ObjectId commitId) throws
         final int initialBuildNumber = project.getLastBuild().getNumber();
         final String commit1 = ObjectId.toString(commitId);
 
-        final int port = server.getConnectors()[0].getLocalPort();
-        if (port < 0) {
-            throw new IllegalStateException("Could not locate Jetty server port");
-        }
-        final String notificationPath = "http://localhost:" + Integer.toString(port)
-                + "/git/notifyCommit?url=" + testRepo.gitDir.toString() + "&sha1=" + commit1;
+        final String notificationPath = rule.getURL().toExternalForm()
+                + "git/notifyCommit?url=" + testRepo.gitDir.toString() + "&sha1=" + commit1;
```

Comment:
BTW `.toExternalForm()` and `.toString()` can both be omitted in the context of string concatenation.